exams/utils.py

```diff
@@ -238,11 +262,7 @@ def authorize_for_latest_passed_course(mmtrack, course_id):
     for enrollment in enrollments_qset:
         # only latest passed course_run per course allowed
         edx_course_key = enrollment.course_run.edx_course_key
-        has_paid_and_passed = (
-            mmtrack.has_passed_course(edx_course_key) and
-            mmtrack.has_paid(edx_course_key)
-        )
-        if has_paid_and_passed:
+        if _has_passed_course(mmtrack, edx_course_key) and _has_paid_for_exam(mmtrack, edx_course_key):
```

Comment:
@pdpinch 
- yes `SUPPRESS_PAYMENT_FOR_EXAM` flag is only to disable payment check.
- user must pass the course to get exam authorization happen.